### PR TITLE
Add ast design and asdl file for SeedAST

### PR DESCRIPTION
**Pull request by codingpotato**
Suggest to use Zephyr Abstract Syntax Definition Language as the specification of SeedAST. The final C# code of SeedAST can be written by hand or generated from ASDL file by a tool.

### CODE REVIEW

File: design/ast.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# SeedAST
+
+The SeedAST (Seed Abstract Syntax Tree) is a high-level representation of the
+SeedBlock program structure. It can be thought of as an abstract representation
```
Comment (wixette) [inline design/ast.md:4]: Not only for SeedBlock, but also for all SeedX languages. SeedAST defines a common subset among SeedBlock and all SeedX languages.

File: grammars/SeedAST.asdl [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- ASDL's 4 builtin types are:
```
Comment (wixette) [inline grammars/SeedAST.asdl:1]: Copyright info and Apache license head, please.

File: design/ast.md (hunk 1 above)
Comment (codingpotato) [reply design/ast.md]: Done

File: grammars/SeedAST.asdl (hunk 2 above)
Comment (codingpotato) [reply grammars/SeedAST.asdl:17]: Done

**Comment by wixette** [comment]
LGTM

**Review by wixette — APPROVED** [review]

**Pull request merged**